src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -62,6 +62,76 @@ std::shared_ptr<IPassManager> BasePass::getPassManager() {
     return sharedMgr;
 }
 
+static auto DoNotSkip = [](CNNLayerPtr) {
+    return false;
+};
+
+/**
+ * @brief Return true if the layer has max one non-1 dimension
+ * (because we can treat it as a single dimension layer, then)
+ */
+static bool IsOneDimLayer(CNNLayerPtr layer) {
+    auto dims = layer->insData[0].lock()->getDims();
+    return std::count_if(std::begin(dims), std::end(dims), [](size_t dim) { return dim > 1; }) <= 1;
+}
+
+/**
+ * @brief Remove 'toRemove' layer from between two other layers - l1 and l2
+ */
+static bool RemoveLayer(CNNLayerPtr l1, CNNLayerPtr toRemove, CNNLayerPtr l2, int l2InputNo = 0) {
```

Comment:
Have you tried to find some existing useful functions in gna_graph_tools.hpp? Or extend them?

Reply:
Yes, there is a method CNNNetworkRemoveLayer, but it does not connect l1 and l2 after removal. I have renamed RemoveLayer to CNNRemoveAndConnect and moved to gna_graph_tools.hpp.